src/Views/CategoryView.vala

```diff
@@ -216,16 +222,18 @@ public class Slingshot.Widgets.CategoryView : Gtk.EventBox {
         category_switcher.select_row (new_selected ?? category_switcher.get_row_at_index (0));
     }
 
-    public void show_filtered_apps (string category) {
-        foreach (unowned Gtk.Widget child in listbox.get_children ()) {
-            child.destroy ();
-        }
-
-        foreach (Backend.App app in view.app_system.apps[category]) {
-            listbox.add (new AppListRow (app.desktop_id, app.desktop_path));
+    [CCode (instance_pos = -1)]
+    private bool filter_function (AppListRow row) {
+        unowned CategoryRow category_row = (CategoryRow) category_switcher.get_selected_row ();
+        if (category_row != null) {
+            foreach (Backend.App app in view.app_system.apps[category_row.cat_name]) {
```

Comment:
I think this can be `unowned Backend.App`?  It is in a similar iteration elsewhere.

Reply:
Looks like no, I get `Invalid assignment from owned expression to unowned variable`

Reply:
Sorry, in the other case the loop was iterating a list returned from a function `get_apps_by_name` but you are accessing the Gee.ArrayList directly.